src/crypto/hashing.rs

```diff
@@ -1 +1,76 @@
 
+use data_encoding::HEXUPPER;
+use sha2::{Sha256, Digest};
+use std::convert::TryInto;
+
+#[derive(Debug, Clone, Copy)]
```

Comment:
May be worth deriving / implementing [std::cmp::Eq](https://doc.rust-lang.org/std/cmp/trait.Eq.html)? This will allow us to easily check if two hashes are equal.